### DISCUSSION TITLE
Arrow key adjustments

### DISCUSSION THREAD

**Discussion by CharlieDixon** [rxhanson/Charmstone-Community#7] [Ideas]
At the moment, when using arrow keys to select an option, you have to return to centre before you can move to the next direction. For example, if I have finder on the upward direction selected, and I want to instead select safari to the left, I currently have to go down and then left. This is a minor detail of course but doesn't feel that intuitive to me.

Also, it would be nice to have the hover feature option apply to the key directions too.

**Comment by rxhanson** [comment]
> At the moment, when using arrow keys to select an option, you have to return to centre before you can move to the next direction. For example, if I have finder on the upward direction selected, and I want to instead select safari to the left, I currently have to go down and then left. This is a minor detail of course but doesn't feel that intuitive to me.

I agree. If this were implemented, do you have any thoughts on how to allow the user to back out and not select an app if they change their mind and decide not to switch apps?

> Also, it would be nice to have the hover feature option apply to the key directions too.

Good point. I'll tack this one on the backlog.

**Reply by CharlieDixon** [reply]
Hmm, just thinking about it now, the first thing that springs to mind is something similar to a joystick-type interface. Not sure on the graphics front how that would look, but it could work by pressing an arrow key to highlight a particular option, and if you release that arrow key it automatically returns to centre. 

The Equip Wheel mod in the game Valheim has something similar to what I have in mind mechanic-wise. In this [picture](https://staticdelivery.nexusmods.com/mods/3667/images/536/536-1615910865-2068956149.jpeg), you can see how the wheel is brought up (by a user key press) and how you can select an item by pointing the joystick in its direction and releasing it while pointing in that direction.

So:
⌘ + ⌥ + ↑ = highlights item
Letting go of all three keys would select the highlighted item.
Letting go of just the arrow key would return to centre/not highlight any option.
Once at centre, the user has the option of releasing ⌘ + ⌥ to exit the interface or the option to press a different arrow to highlight another option.

**Reply by rxhanson** [reply]
I like this direction. Here are some initial thoughts on it.
1. Reaching the far slots would require something like timing adjustments (hold for x millis to go to far), or double tap arrow key to get there. Both of those options would require some kind of communication to the user. Maybe there's a better idea for that?
2. There's something about having to hold the shortcut and release all the keys of the shortcut that just doesn't feel nice. Maybe it would feel comfortable if I were used to it, but it's just not quite as approachable feeling. 

Another alternative is to have a cancel keyboard shortcut. Arrows skip the middle, and  `⌘ + ⌥ + Z` or `⌘ + ⌥ + "numpad 0"` would cancel it. Still more complicated than what it is now, which makes me not as excited about the prospect of adding it in, but just thinking out loud.
